OpenBCISample.js

```diff
@@ -0,0 +1,200 @@
+/*
+Author: AJ Keller
+*/
+
+/*
+Constants for interpreting the EEG data
+*/
+// Reference voltage for ADC in ADS1299.
+//   Set by it's hardware.
+const ADS1299_VREF = 4.5;
+// Assumed gain setting for ADS1299.
+//   Set by it's Arduino code.
+const ADS1299_GAIN = 24.0;
+// Start byte
+const BYTE_START = 0x0A;
+// Stop byte
+const BYTE_STOP	= 0xC0;
+// For conversion of Volts to uVolts
+const CONVERT_VOLTS_TO_MICROVOLTS = 1000000;
+// The sample rate in Hz
+const SAMPLE_RATE = 250.0;
+// Scale factor for aux data
+const SCALE_FACTOR_ACCEL = 0.002 / Math.pow(2,4);
+// Scale factor for channelData
+const SCALE_FACTOR_CHANNEL = ADS1299_VREF / ADS1299_GAIN / (Math.pow(2,23) - 1);
+
+var k = require('./OpenBCIConstants');
+
+/*
+ Errors
+ */
+const kErrorInvalidByteLength = "Invalid Packet Byte Length";
+const kErrorInvalidByteStart = "Invalid Start Byte";
+const kErrorInvalidByteStop = "Invalid Stop Byte";
+
+module.exports = {
+    convertPacketToSample: function (dataBuf) {
+        var self = this;
+        if(dataBuf === undefined || dataBuf === null) {
+            reject('data is undefined');
+        }
+        var numberOfBytes = dataBuf.byteLength;
+        var scaleData = true;
+
+        if (dataBuf[0] != BYTE_START) return;
```

Comment:
Ok. So no this does not return an error message. The 'reject('data is undefined'); should be just 'return;' Should we do an error return?